### ISSUE TITLE
Players on vacation in certain games can crash the client

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#83]
Playing with Nancy, Grizz and Charles, the client kept locking up on me repeatedly and it didn't really matter which browser I used. The JavaScript Console sometimes had mystery complaints in it:

`Error: Unreachable code should not be executed (evaluating 'Module.STDWEB_PRIVATE.dyncall( "vii", adapter_pointer, [function_pointer, args] )')`

there is a chance that the version of rust-nightly that I've been using has issues. I have not seen this problem before and I do tend to upgrade to the latest nighty every day, which I need to stop doing.

I also should switch to web-sys.

**Comment by ctm** [comment]
I was able to reproduce a similar error and its amazingly embarrassing that I didn't realize what was happening.

The biggest thing is that I do almost all my testing locally with a non-release build primarily because it takes a long time to compile release builds. However, release builds don't check for underflow and apparently if you have a ridiculously big allocation you can get that "Unreachable code" error. So this is the code that caused the problem:
```
        // The indentation hack in here is really annoying.  I'm not
        // proud of how I wrote it, but it works, and will probably be
        // going away.
        use mb2_text::player::{CARDS_PREFIX, CARDS_PREFIX_LEN};
        let mut line = line;
        let new_line;
        if line.starts_with(CARDS_PREFIX) {
            new_line = format!(
                "{empty:width$}{line}",
                empty = "",
                width = self.last_indent as usize - CARDS_PREFIX_LEN,
                line = line,
            );
```
The problem is that when people are on vacation, `last_indent` can be less than `CARDS_PREFIX_LEN` which results in underflow which results in a very large allocation. That section of code is a known hack that I had planned on getting rid of "soon", but other features and whatnot jumped the queue. 

Although I do a bunch of testing locally, I hadn't tested vacation code in a while, so I hadn't hit that in a non-release setting.

If I had my wits, I would have realized what was going on, but I just haven't had enough experience with crashes running release builds because I use them so infrequently.

FWIW, I'll continue deploying release builds to devctm.com because I think I now will recognize that sort of error. The backtrace really did give me enough information to make an informed guess, it's just that I didn't look far enough into it to realize what was happening. Heck, the first tine everything died, I thought the server had crashed, because when I closed my table window the new table window came up blank. That was just because the replay code triggered the same bug.

**Comment by ctm** [comment]
Fixed and deployed, although I did find #84 while debugging this one. It's always something.
